pkg/tc/checker.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2019-2020 vChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tc
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"github.com/codenotary/immudb/pkg/api/schema"
+	"github.com/codenotary/immudb/pkg/client"
+	"github.com/codenotary/immudb/pkg/logger"
+	mrand "math/rand"
+	"time"
+)
+
+// ErrConsistencyFail happens when a consistency check fails. Check the log to retrieve details on which element is failing
+const ErrConsistencyFail = "consistency check fail at index %d"
+
+type immuTc struct {
+	Client  client.ImmuClient
+	Logger  logger.Logger
+	Quit    bool
+	Trusted bool
+}
+
+// ImmuTc trust checker interface
+type ImmuTc interface {
+	Start(context.Context) (err error)
+	Stop(context.Context)
+	GetStatus(context.Context) bool
+}
+
+// NewImmuTc returns new trust checker service
+func NewImmuTc(c client.ImmuClient, l logger.Logger) ImmuTc {
+	return &immuTc{c, l, false, true}
+}
+
+// Start start the trust checker loop
+func (s *immuTc) Start(ctx context.Context) (err error) {
+	s.Logger.Infof("Start scanning ...")
+	return s.checkLevel0(ctx)
+}
+
+// Stop stop the trust checker loop
+func (s *immuTc) Stop(ctx context.Context) {
+	s.Quit = true
+}
+
+// GetStatus return status of the trust checker. False means that a consistency checks was failed
+func (s *immuTc) GetStatus(ctx context.Context) bool {
+	return s.Trusted
+}
+
+func (s *immuTc) checkLevel0(ctx context.Context) (err error) {
+	for ok := true; ok; ok = !s.Quit {
```

Comment:
you can make s.Quit channel and make a select loop it with time.Tick(10*time.Second). So in any quit case you can remove this 10 seconds barrier.